functions/src/index.ts

```diff
@@ -1,15 +1,59 @@
 import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
 import * as express from 'express';
 
+// FIX: need to initialize Firebase prior to import of app.
+// there's probably a better way to fix this issue.
+admin.initializeApp();
+
+import * as bodyParser from 'body-parser';
+import { RegisterRoutes } from '../build/routes';
+
+import * as openapi from '../build/swagger.json';
+
+// THIS FILE SHOULDN'T RUN WITHOUT FIREBASE EMULATOR!
+// nor is it meant to be. use server.ts!
+
+export const app = express();
```

Comment:
shouldn't this just import app from app.ts and use that? (ie. avoid this code duplication)

Reply:
yeahhhhhhhhhhhhhhhhhhhhhhhhhh so i like broke the app last night and I realized the middle-ware for rewriting URLs has to be loaded in before the routes and what no so I decided for now, till there's better tooling to differentiate run-time environments.